### DISCUSSION TITLE
[Request for Documentation] I have many many questions.

### DISCUSSION THREAD

**Discussion by jpiabrantes** [oddbit/flutter_facebook_app_events#97] [General]
I am trying to setup facebook events in iOs to get the automatic events: app install, launches, purchases.

Do I only need to install this package?

Do I need to `FacebookAppEvents();` when the app launches?

Can I ignore step 3 and 6 on [Getting Started with App Events for iOS](https://developers.facebook.com/docs/app-events/getting-started-app-events-ios)?

On step 4 the Bundle ID is always empty no matter how many times I fill it in and press save.

If I make an in-app purchase in sandbox mode, will it send an event to facebook?

What's the best way to test if events are going through? How long do they take to arrive at facebook?

I think that everyone would win to have these questions written on the documentation.

Thank you very much for this package!

**Comment by fransiskapw** [comment]
Hi there,

I just installed the package and was able to see live test events successfully in iOS (shown in `https://business.facebook.com/events_manager2/list/app/<app-id>/test_events?business_id=<business-id>`). Though I wasn't able to see live Android event, it updated 'Last Android Install' here: `https://developers.facebook.com/tools/app-ads-helper/?id=<app-id>`, so I hope it works as well for Android

I'm answering based on my limited knowledge:

**Do I only need to install this package?**
Yes

**Do I need to `FacebookAppEvents();` when the app launches?**
Yes, see the example for reference: https://pub.dev/packages/facebook_app_events/example

**Can I ignore step 3 and 6 on Getting Started with App Events for iOS?**
Yes

**On step 4 the Bundle ID is always empty no matter how many times I fill it in and press save.**
You can edit directly on ios/Runner/Info.plist
Though for the Info.plist, don't add additional `<array>` if you already have `CFBundleURLTypes`
What's added to my Info.plist:
`<dict>
  <key>CFBundleURLSchemes</key>
    <array>
      <string>fb{app-id}</string>
    </array>
</dict>`

**If I make an in-app purchase in sandbox mode, will it send an event to facebook?**
Don't know, you should check Facebook Documentation on this

**What's the best way to test if events are going through? How long do they take to arrive at facebook?**
You can test live event on Event Manager
`https://business.facebook.com/events_manager2/list/app/<app-id>/test_events?business_id=<business-id>
`
Or check here for implementation status
`https://developers.facebook.com/tools/app-ads-helper/?id=<app-id>
`

Hopefully, it helps!
